### DISCUSSION TITLE
Install mode instead of keep the app running

### DISCUSSION THREAD

**Discussion by nhac-lly** [araghon007/X1nput#40] [Ideas]
AFAIK, there are too many versions of XInput to keep in track, and the program is here to find the correct hook for it. 
But can we have the option to just install that version of XInput so that we don't to hook it everytime we run a game? Since most the game to use this on no longer have update to change the version of XInput, I think it's ok to just install once.

**Comment by araghon007** [comment]
Yeah, sorry for the confusion, and I'm afraid some changes I'm planning to do will only add to that. There's really only 2 versions of X1nput. 
The old one worked as a "replacement" for xinput.dll that required the X1nput dll (and you'd also need to know if it was 32 or 64-bit) to be duplicated and renamed to all the versions of xinput that are out there.
The new one is much smarter. It's using minhook, which allowed me to hook onto xinput functions directly using a single dll (still needs separate one for 32 and 64-bit though), and it also allows me to bypass xinput altogether and just use device IO control, which also allows me to support Steam's own driver.

The configurator only serves to change the config file and inject the 32-bit dll into 32-bit processes, and 64-bit dll into 64-bit processes.
I could either add a way for the configurator to auto-inject into selected games on launch, or go the way some game mods do and have some dummy dll that always gets loaded by the target process, which would make it possible to use without the configurator. The issue with that approach is that it would be difficult to make it universal, as different games load different sets of dlls.

**Comment by GGdesu** [comment]
I know the discussion isn't about that, but I decided to ask here because it's the most recent answer I saw in the discussion section.

Regarding X1nput, in games that support impulsive trigger, does it activate this native support? like the vibrations of the triggers and their intensities are independent of each other?

In games that don't support the impulsive trigger, it only imitates the traditional vibrations of the rotor motor, right?
